public/app/features/alerting/unified/mocks/server/events.ts

```diff
@@ -21,3 +22,46 @@ export function waitForServerRequest(handler: HttpHandler) {
     });
   });
 }
+
+interface SerializedRequest {
+  method: string;
+  url: string;
+  body: string | JsonValue;
+  headers: string[][];
+}
+
+export async function captureRequests(): Promise<Request[]> {
+  let requests: Request[] = [];
+
+  server.events.on('request:start', ({ request }) => {
+    requests.push(request);
+  });
+
+  return new Promise(async (resolve) => {
+    resolve(requests);
+  });
```

Comment:
I just checked and I think you can actually remove the promise logic from here and just return `requests`. The problem before was that we weren't starting to listen for events at the right time?

Reply:
Correct, I don't think we need this bit! :D